### PR TITLE
Many repl enhancements including command name parsing and general testing

### PR DESCRIPTION
**Pull request by ice1000**
As title

### CODE REVIEW

File: cli/src/test/java/org/aya/cli/JlineReplTest.java [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+    assertEquals("zero", parser.parse("suc     zero", 5).word());
+    assertEquals("", parser.parse("suc  zero      ", 12).word());
+  }
+
+  @Test public void parenTyCode() {
+    var line = parser.parse("(Ty", 2);
+    var candidates = new ArrayList<Candidate>();
+    AyaCompleters.KW.complete(null, line, candidates);
+    assertFalse(candidates.isEmpty());
+  }
```
Comment (imkiva) [inline cli/src/test/java/org/aya/cli/JlineReplTest.java:63]: Impressive!

**Review by imkiva — APPROVED** [review]
So quick! Oh my ice1000!

File: cli/src/test/java/org/aya/cli/JlineReplTest.java (hunk 1 above)
Comment (ice1000) [reply cli/src/test/java/org/aya/cli/JlineReplTest.java:63]: Isn't you who implemented this? LOL

File: cli/src/main/java/org/aya/cli/repl/jline/completer/CmdCompleter.java [hunk 2]
```diff
@@ -36,10 +36,9 @@ public void complete(LineReader reader, ParsedLine line, List<Candidate> candida
     }
     var trim = line.line().trim();
     if (trim.startsWith(Command.PREFIX)) {
-      // TODO: replace with Option.mapNotNull
-      cmd.parse(trim.substring(1)).command()
-        .flatMap(c -> Option.of(c.argFactory()))
-        .flatMap(arg -> Option.of(arg.completer()))
+      cmd.parse(trim.substring(1)).command().view()
+        .mapNotNull(CommandManager.CommandGen::argFactory)
+        .mapNotNull(CommandArg::completer)
```
Comment (imkiva) [inline cli/src/main/java/org/aya/cli/repl/jline/completer/CmdCompleter.java:41]: I think we should check there's only one completer.

**Comment by ice1000** [comment]
bors r+

M for Mario, no
W for Waluigi, yay~

File: cli/src/main/java/org/aya/cli/repl/jline/completer/CmdCompleter.java (hunk 2 above)
Comment (ice1000) [reply cli/src/main/java/org/aya/cli/repl/jline/completer/CmdCompleter.java:41]: Why? For several completers we just join all of them

Comment (imkiva) [reply cli/src/main/java/org/aya/cli/repl/jline/completer/CmdCompleter.java:41]: Well ok. I think it's fine now.

Comment (imkiva) [reply cli/src/main/java/org/aya/cli/repl/jline/completer/CmdCompleter.java:41]: My head is temporarily offline

**Pull request merged**